code/parse/sexp.cpp

```diff
@@ -1820,11 +1820,6 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, i
 	if (!check_operator_argument_count(count, op))
 		return SEXP_CHECK_BAD_ARG_COUNT;  // incorrect number of arguments
 
-	// Goober5000 - if this is a list of stuff that has the special argument as
-	// an item in the list, assume it's valid
-	if (special_argument_appears_in_sexp_list(op_node))
-		return 0;
-
```

Comment:
Removing this check breaks the dreamscape, Her Finest Hour, and Universal Truth from BP2 (and presumably other missions in other mods). Whenever FRED saves a mission that uses `modify-variable` to set a number variable to `<argument>`, it treats that second node as being `SEXPT_NUMBER` and therefore `SEXP_ATOM_NUMBER`, meaning it gets written to the mission without quotation marks. When it comes time to load the mission, the lack of this check means that it goes on to the `SEXP_ATOM_NUMBER` validation code below, which detects that `<argument>` is not a number and returns `SEXP_CHECK_INVALID_NUM`. This makes FSO refuse to load not only missions that used to work fine before this PR was merged, but also missions made using current FRED2 (and, presumably, qtFRED).